js/style/style.js

```diff
@@ -22,8 +22,9 @@ var getWorkerPool = require('../global_worker_pool');
 
 module.exports = Style;
 
-function Style(stylesheet, animationLoop, options) {
-    this.animationLoop = animationLoop || new AnimationLoop();
+function Style(stylesheet, map, options) {
+    this.map = map;
```

Comment:
Removing the dependency was my first instinct as well. Unfortunately it wasn't straightforward to do so and will require a standalone PR. 

Even though the dependency is undesirable, it is a real dependency and should be passed explicitly and simply. As our code is architected, `Source`s need to know about the state of `Camera` (which is a mixin on `Map`). I see #2741 as the proper long-term fix.